### PR TITLE
Tools: support lldb debugging for sim_vehicle and autotest

### PR DESCRIPTION
**Pull request by andyp1per**
lldb generally works much, much better than gdb on macOS

### CODE REVIEW

**Comment by andyp1per** [comment]
@peterbarker FYI

File: Tools/autotest/sim_vehicle.py [hunk 1]
```diff
@@ -842,6 +855,14 @@ def generate_frame_help():
                      action='store_true',
                      default=False,
                      help="use gdb for debugging ardupilot (no auto-start)")
+group_sim.add_option("-X", "--lldb",
```
Comment (peterbarker) [inline Tools/autotest/sim_vehicle.py]: Could we do without the "-X" and "-x" here, please?

Comment (andyp1per) [reply Tools/autotest/sim_vehicle.py]: Yeah, "b" was taken. Any suggestions? Or just have the long options?

Comment (andyp1per) [reply Tools/autotest/sim_vehicle.py]: Ok removed

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**